### PR TITLE
Added upgrade guide for CrateDB 3.0.0

### PR DESCRIPTION
**Pull request by Autophagy**
As part of this PR, I've also removed the upgrade notes from 3.0.0 since it was duplicated information, and this makes the 3.0.0 release notes specifically about the release itself, and not the upgrade process.

What do you think? I'm unsure how much detail to add to the upgrade guide - how to upgrade specific distributions, specific stuff for docker, etc?

### CODE REVIEW

**Comment by mikethebeer** [comment]
@Autophagy tbh I like the idea 👍

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+.. _3.0.0_upgrade_guide:
+
+=======================
+Crate 3.0 Upgrade Guide
+=======================
+
+Upgrade Strategy
+================
+
+Upgrading to 3.0.0 is only possible from a 2.X.Y release. It is advisable to to
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: Should read: `3.0.x` and `2.x`

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+try and upgrade to the latest 2.3 release before moving to 3.0.0. Tables
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: 3.0.x

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+that were created prior to 2.0.0 have been **deprecated**, and must be upgraded
+before moving to 3.0.0. This can be done by using `COPY TO` and `COPY FROM`
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: inline code formatting requires double-backticks

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+Once this is done, you must perform a `Full Cluster Restart`_, where all nodes
+are shut down, upgraded and started again. This is contrast to a rolling restart
+upgrade strategy, which if performed between major versions may lead to crashes,
+data corruption or unexpected behaviour. You should also take care to `backup`_
+your existing data before upgrading.
+
+.. note::
+
+   If using docker, docker service update will perform a rolling restart. To do
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: Docker is a proper noun and should start with uppercase letter.

Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: `docker service upgrade` is a command and should be formatted as inline code

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst (hunk 1 above)
Comment (chaudum) [reply blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: Please check all occurrences. Writing `3.0.x` will make keep the upgrade guide valid when there is a new 3.0 hotfix.

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+few other changes that you must be aware of from the application side.
+
+- The ``CREATE REPOSITORY`` statement for creating backup repositories has been
+  changed. When using Amazon S3 for backup storage, you previously had to specify
+  the region setting in order to determine the AWS region where the bucket was
+  located. This region is now inferred automatically, but if you still need to
+  specify this, use the endpoint setting to specify the AWS API endpoint.
+
+- The schema for HTTP authentication has been changed. Previously, you were
+  able to use the ``X-User header`` to provide a username. This has now been
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: `X-User` HTTP header

File: blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  deprecated in favour of the standard `HTTP Authorization header`_.
+
+- The ``_node`` column in ``sys.shards`` and ``sys.operations`` has been
+  renamed to node, but has been trimmed down only to include the id and name of
+  the node (``node[‘id’]`` and ``node[‘name’]``). In order to get the full node
```
Comment (chaudum) [inline blackbox/docs/appendices/upgrade-guides/3.0.0-upgrade-guide.rst]: `id` should be quoted with single quote, not tick

**Comment by Autophagy** [comment]
I've made the upgrade notes part of the 3.0.0 notes, based on @nslater's feedback. Also addressed your points @chaudum.

**Pull request merged**